scripts/utils.sh

```diff
@@ -0,0 +1,384 @@
+  echo "$(date +"$DATE_FORMAT") | ACTION - $1"
+}
+
+#######################################
+# Echo a text string using the ERROR format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDERR.
+#######################################
+function utils::echo_error() {
+  echo "$(date +"$DATE_FORMAT") | ERROR  - $1" >&2
+}
+
+#######################################
+# Echo a text string using the INFO format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
+#######################################
+function utils::echo_info() {
+  echo "$(date +"$DATE_FORMAT") | INFO   - $1"
+}
+
+#######################################
+# Echo a text string using the TITLE format.
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
+#######################################
+function utils::echo_title() {
+  echo ""
+  echo "###############################################################################"
+  echo "$1"
+  echo "###############################################################################"
+}
+
+#######################################
+# Harden Apache2 Webserver settings.
+# Arguments:
+#   Apache2 security configuration file to update, a file path.
+# Outputs:
+#   Writes normal log messages to STDOUT.
+#   Writes error messages to STDERR.
+#######################################
+function utils::harden_apache2() {
+  local config_file_path="$1"
+
+  utils::update_apache2_config_file "ServerTokens" "Prod" "${config_file_path}"
+  utils::update_apache2_config_file "ServerSignature" "Off" "${config_file_path}"
+}
+
+#######################################
+# Mount a data disk identified by its size to a mount point.
+# Since that is no way to predict the device name of a data disk, we use the
+# device size as to identify the data disk. This means that is it impossible
+# to use this function utils::to mount a data disk that has the same size as another
+# device attached to the server.
+# The data disk is formated if no file system exists.
+# Globals:
+# Arguments:
+#   1) size of the disk to mount, a string as returned by the
+#      "lsblk --output name,size" command.
+#   2) mount_point_path, a path
+# Outputs:
+#   Writes normal log messages to STDOUT.
+#   Writes error messages to STDERR.
+#######################################
+function utils::mount_data_disk_by_size() {
+  local data_disk_size="$1"
+  local data_disk_mount_point_path="$2"
+
+  readonly DEFAULT_FILE_SYSTEM_TYPE="ext4"
+  readonly FSTAB_FILE_PATH="/etc/fstab"
+  readonly TIMEOUT=60
+
+  local data_disk_block_path
+  local data_disk_file_system_type
+  local data_disk_file_system_uuid
+  local elapsed_time
+
+  utils::echo_action 'Retrieving data disk block path using data disk size as index...'
+  data_disk_block_path="/dev/$(lsblk --noheadings --output name,size | awk "{if (\$2 == \"${data_disk_size}\") print \$1}")"
```

Comment:
Are the Double quotes causing problems in the variable interpretation?

Reply:
No. The quotes are wrapped inside $(...) which kind of "resets" the quote sequence.